resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/submitsteps/hadoopsteps/HadoopStepsOrchestrator.scala

```diff
@@ -0,0 +1,104 @@
+  namespace: String,
+  hadoopConfigMapName: String,
+  submissionSparkConf: SparkConf,
+  hadoopConfDir: String) extends Logging{
+   private val isKerberosEnabled = submissionSparkConf.get(KUBERNETES_KERBEROS_SUPPORT)
+   private val maybePrincipal = submissionSparkConf.get(KUBERNETES_KERBEROS_PRINCIPAL)
+   private val maybeKeytab = submissionSparkConf.get(KUBERNETES_KERBEROS_KEYTAB)
+     .map(k => new File(k))
+   private val maybeExistingSecret = submissionSparkConf.get(KUBERNETES_KERBEROS_DT_SECRET_NAME)
+   private val maybeExistingSecretItemKey =
+     submissionSparkConf.get(KUBERNETES_KERBEROS_DT_SECRET_ITEM_KEY)
+   private val maybeRenewerPrincipal =
+     submissionSparkConf.get(KUBERNETES_KERBEROS_RENEWER_PRINCIPAL)
+   private val hadoopConfigurationFiles = getHadoopConfFiles(hadoopConfDir)
+   private val hadoopUGI = new HadoopUGIUtil
+   logInfo(s"Hadoop Conf directory: $hadoopConfDir")
+
+   require(maybeKeytab.forall( _ => isKerberosEnabled ),
+     "You must enable Kerberos support if you are specifying a Kerberos Keytab")
+
+   require(maybeExistingSecret.forall( _ => isKerberosEnabled ),
+     "You must enable Kerberos support if you are specifying a Kerberos Secret")
+
+   OptionRequirements.requireBothOrNeitherDefined(
+     maybeKeytab,
+     maybePrincipal,
+    "If a Kerberos keytab is specified you must also specify a Kerberos principal",
+     "If a Kerberos principal is specified you must also specify a Kerberos keytab")
+
+   OptionRequirements.requireBothOrNeitherDefined(
+     maybeExistingSecret,
+     maybeExistingSecretItemKey,
+     "If a secret storing a Kerberos Delegation Token is specified you must also" +
+     " specify the label where the data is stored",
+     "If a secret data item-key where the data of the Kerberos Delegation Token is specified" +
+       " you must also specify the name of the secret")
+
+  def getHadoopSteps(): Seq[HadoopConfigurationStep] = {
+    val hadoopConfBootstrapImpl = new HadoopConfBootstrapImpl(
+      hadoopConfigMapName,
+      hadoopConfigurationFiles,
+      hadoopUGI)
+    val hadoopConfMounterStep = new HadoopConfMounterStep(
+      hadoopConfigMapName,
+      hadoopConfigurationFiles,
+      hadoopConfBootstrapImpl,
+      hadoopConfDir)
+    val maybeKerberosStep =
+      if (isKerberosEnabled) {
+        maybeExistingSecret.map(existingSecretName => Some(new HadoopKerberosSecretResolverStep(
+         submissionSparkConf,
+          existingSecretName,
+          maybeExistingSecretItemKey.get))).getOrElse(Some(
+            new HadoopKerberosKeytabResolverStep(
+              submissionSparkConf,
+              maybePrincipal,
+              maybeKeytab,
+              maybeRenewerPrincipal,
+              hadoopUGI)))
+      } else {
+        Option.empty[HadoopConfigurationStep]
+      }
+    Seq(hadoopConfMounterStep) ++ maybeKerberosStep.toSeq
+  }
+
+  private def getHadoopConfFiles(path: String) : Seq[File] = {
```

Comment:
Should we filter out `.xml` files only?

Reply:
I mean the goal is to mount all files in the hadoop conf directory. `.xml` or not. But if we wish to filter we can do that as well